### ISSUE TITLE
RTL may report incorrect N and Z flag

### ISSUE THREAD

**Issue by guptasanchi** [SimpleCPU/SimpleCPU#17]
While debugging one of the failures, we found out that the RTL uses the output of the adder to calculate the Z and the N flag. This is shown in the following code as well - 
```
    assign  z_alu               = adder_out_alu ? 1'b0 : 1'b1;
    assign  n_alu               = adder_out_alu[31];
```

The N and Z flags should be calculated based on the output of the ALU rather than using the output of the adder.

**Comment by adityagupta1011** [comment]
A potential fix to this could be by using the res_alu value instead of the adder_out_alu signal.
@raulbehl what do you suggest?

**Comment by guptasanchi** [comment]
@adityagupta1011 - Precisely. I have checked in the changes to do just what you suggested. Thank you for looking into it. Fix in 1b12c39

**Comment by mohitgoyal2011** [comment]
I think adityagupta1011 is right.

**Comment by guptasanchi** [comment]
@mohitgoyal2011 - Yes. :)

**Comment by raulbehl** [comment]
@adityagupta1011 - Precisely. I have checked in the changes to do just what you suggested. Thank you for looking into it. Fix in 1b12c39
@mohitgoyal2011 - Yes. His suggestion was right.
